rust/relay/src/server.rs

```diff
@@ -107,18 +108,41 @@ pub enum Command {
         family: AddressFamily,
     },
 
-    ForwardData {
+    ForwardDataClientToPeer {
         id: AllocationId,
         data: ClientToPeer,
         receiver: PeerSocket,
     },
+
     /// At the latest, the [`Server`] needs to be woken at the specified deadline to execute time-based actions correctly.
     Wake { deadline: SystemTime },
 }
 
 #[derive(Debug, PartialEq)]
 pub struct ClientToPeer(ChannelData);
 
+#[derive(Debug, PartialEq)]
+pub struct PeerToClient {
+    buf: BytesMut,
+}
+
+impl PeerToClient {
+    pub fn new(msg: &[u8]) -> Self {
+        let mut buf = BytesMut::zeroed(msg.len() + 4);
```

Comment:
We still allocate here but we leave enough space to put the channel data header in front of it.